### PR TITLE
Registry certs/marriage add refactor

### PR DESCRIPTION
**Pull request by jessicamarcus**
- refactors out some components that can be shared between Marriage and Birth
- reorganizes some Stories for Storybook
- adds needed files and components for Marriage

Please note that this PR is fairly preliminary, but I didn't want to go too long without committing...

### CODE REVIEW

File: services-js/registry-certs/client/birth/CheckoutPage.stories.tsx [hunk 1]
```diff
@@ -1,35 +1,26 @@
...
+
 import CheckoutDao from '../dao/CheckoutDao';
-import OrderProvider from '../store/OrderProvider';
 import Order, { OrderInfo } from '../models/Order';
+import OrderProvider from '../store/OrderProvider';
 import BirthCertificateRequest from '../store/BirthCertificateRequest';
 
+import CheckoutPage from './CheckoutPage';
+
+import { TYPICAL_REQUEST as birthCertRequest } from '../../fixtures/client/birth-certificates';
```
Comment (fionawhim) [inline services-js/registry-certs/client/birth/CheckoutPage.stories.tsx:13]: 👍

File: services-js/registry-certs/client/common/ReviewCertificateRequest.tsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    const { certificateType, siteAnalytics } = this.props;
+
+    this.props.certificateRequest.clearCertificateRequest();
+
+    siteAnalytics.sendEvent('user reset', {
+      category: capitalize(certificateType),
+      label: 'start over',
+    });
+
+    Router.push(`/${certificateType}`);
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/ReviewCertificateRequest.tsx:91]: Be aware that this is here; you may want to update this to be explicit about the separate routes. Right now this is a hidden dependency that certificateTypes and page routes match exactly.

File: services-js/registry-certs/client/common/checkout/OrderConfirmationContent.tsx [hunk 3]
```diff
@@ -64,11 +68,14 @@ export default function BirthConfirmationContent({
 
           <p className="t--info" style={{ fontStyle: 'normal' }}>
             Have any questions? Email the Registry Department at{' '}
-            <a href="mailto:birth@boston.gov">birth@boston.gov</a>.
+            <a href={`mailto:${certificateType}@boston.gov`}>
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/checkout/OrderConfirmationContent.tsx:71]: I wonder if you want a map somewhere that’s certificateType -> { email, route, &c. }

File: services-js/registry-certs/client/common/checkout/ReviewContent.tsx [hunk 4]
```diff
@@ -426,7 +405,10 @@ export default class ReviewContent extends React.Component<Props, State> {
             />
             <span className="cb-l">
               I understand that{' '}
-              <strong>{certificatesName} are non-refundable</strong>.
+              <strong>
+                {this.props.certificateType} certificates are non-refundable
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/checkout/ReviewContent.tsx:409]: Same note here, where you're mixing in a symbol (certificateType) with human-readable text

File: services-js/registry-certs/client/marriage/CheckoutPage.tsx [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+      // eslint-disable-next-line no-console
+      console.error(e);
+
+      await Router.push(confirmationUrl);
+
+      window.scroll(0, 0);
+    }
+  };
+
+  sendBirthCertificateProduct() {
```
Comment (fionawhim) [inline services-js/registry-certs/client/marriage/CheckoutPage.tsx:292]: `sendMarriageCertificateProduct`?

**Review by fionawhim — APPROVED** [review]

File: services-js/registry-certs/client/common/checkout/OrderConfirmationContent.tsx (hunk 3 above)
Comment (jessicamarcus) [reply services-js/registry-certs/client/common/checkout/OrderConfirmationContent.tsx:71]: Good thought, thanks! I’ll merge this PR now and revisit this soon.

**Pull request merged**